mkosi/distributions/opensuse.py

```diff
@@ -94,6 +111,42 @@ def zypper_modifyrepo(state: MkosiState, repo: str, caching: bool) -> None:
     invoke_zypper(state, [], "modifyrepo", ["--keep-packages" if caching else "--no-keep-packages"], repo)
 
 
+def zypper_init_repositories(state: MkosiState) -> None:
+    # If we need to use a local mirror, create a temporary repository
+    # definition, which is valid only at image build time. It will be removed
+    # from the image and replaced with the final repositories at the end of the
+    # installation process.
+    #
+    # We need to enable packages caching in any cases to make sure that the
+    # package cache stays populated after "zypper install".
+
+    if state.config.local_mirror:
+        zypper_addrepo(state, state.config.local_mirror, "local-mirror", caching=True)
+    else:
+        for (name, url) in OpensuseInstaller.repositories(state):
+            if state.config.base_image is None:
+                zypper_addrepo(state, url, name, caching=True)
+            else:
+                zypper_modifyrepo(state, name, caching=True)
+
+
+def zypper_finalize_repositories(state: MkosiState) -> None:
+    if state.config.local_mirror:
+        zypper_removerepo(state, "local-mirror")
+
+    # Disable package caching in the image that was enabled previously to
+    # populate mkosi package cache.
+    for (name, url) in OpensuseInstaller.repositories(state):
+        if state.config.local_mirror and state.config.base_image is None:
+            zypper_addrepo(state, url, name)
+
+        elif state.config.base_image is None:
+            zypper_modifyrepo(state, name, caching=False)
+
+        else:
+            zypper_removerepo(state, name)
```

Comment:
```suggestion
        if state.config.local_mirror and state.config.base_image is None:
            zypper_addrepo(state, url, name)
        elif state.config.base_image is None:
            zypper_modifyrepo(state, name, caching=False)
        else:
            zypper_removerepo(state, name)
```
no need for the extra vertical whitespace